### ISSUE TITLE
False positive in `Style/TrailingUnderscoreVariable` cop

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#3380]
When this cop encounters destructuring code like:

```
_, *rest, _ = *arr
```

it incorrectly marks the trailing `_` as an offense, but this is needed to indicate how many tail elements we want to drop when destructuring.

--------

## Expected behavior

RuboCop does not register an offense.

## Actual behavior

```
C: Style/TrailingUnderscoreVariable: Do not use trailing _s in parallel assignment. Prefer _, *rest, = *arr.
_, *rest, _ = *arr
          ^^
```

## Steps to reproduce the problem

Run `bundle exec rubocop --only Style/TrailingUnderscoreVariable`, on:

```
_, *rest, _ = *arr
```

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.0 x86_64-darwin15)
```
